packages/medusa/src/services/product-category.ts

```diff
@@ -235,6 +235,27 @@ class ProductCategoryService extends TransactionBaseService {
       await productCategoryRepository.addProducts(productCategoryId, productIds)
     })
   }
+
+  /**
+   * Remove a batch of product from a product category
+   * @param productCategoryId - The id of the product category on which to remove the products
+   * @param productIds - The products ids to remove from the product category
+   * @return the product category on which the products have been removed
+   */
+  async removeProducts(
+    productCategoryId: string,
+    productIds: string[]
+  ): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const productCategoryRepository: ProductCategoryRepository =
+        manager.getCustomRepository(this.productCategoryRepo_)
+
+      await productCategoryRepository.removeProducts(
+        productCategoryId,
+        productIds
+      )
```

Comment:
todo: Should return the product category
```suggestion
  ): Promise<ProductCategory> {
    return await this.atomicPhase_(async (manager) => {
      const productCategoryRepository: ProductCategoryRepository =
        manager.getCustomRepository(this.productCategoryRepo_)

      await productCategoryRepository.removeProducts(
        productCategoryId,
        productIds
      )
      
      return await this.retrieve(productCategoryId)
```

Reply:
I don't have a strong opinion on this, mainly just for the sake of making it slightly more convenient for the consumer. I see we are using both patterns across the codebase.

Happy to stick with this :)

Reply:
I think what stood out was the `retrieve` invocation, looking through the codebase we do both, and we should align what we do across domains (sales channels, product collections, disc. conditions and pub-api-keys, I might have missed some). I'm okay with doing `retrieve` as long as we do so broadly 👍

Reply:
The idea was that in most cases, returning an entity from within some method like update or delete does not contains what the user expect and therefore the consumer will almost always re do a retrieve after calling the method. Therefore we end up doubling the retrieve because the data are not the expected one. Doing the action and then letting the user retrieve or reload the data with whatever he wants looks more  aligned. This is why with cart, line item etc in some cases we removed the retrieve to return void in favor of the consumer doing the retrieve after

Reply:
Basically, the update is in charge of updating, the delete to delete and that’s it. Then depending on the needs the user can choose to retrieve or reload the entity with the appropriate options. Even us in the code base we don’t rely on the returned data as we almost always re retrieve the entity and it is the same for the users in the community in the majority of the cases 😜 i go back to my vac guys 💪

Reply:
Got it! The performance considerations are reasonable. Let's keep it as is 👍